File: Python/0224_happy_number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        while n != 1 and n != 4:
            n = sum(int(digit) ** 2 for digit in str(n))
        return n == 1

# 測試
solution = Solution()
print(solution.isHappy(19))   # Output: True
print(solution.isHappy(2))    # Output: False
print(solution.isHappy(8397)) # Output: True

File: JavaScript/0224_happy_number.js
function isHappy(n) {
    while (n !== 1 && n !== 4) {
        n = String(n).split('').map(Number).reduce((sum, digit) => sum + digit ** 2, 0);
    }
    return n === 1;
}

// 測試
console.log(isHappy(19));   // Output: true
console.log(isHappy(2));    // Output: false
console.log(isHappy(8397)); // Output: true